File: bench/trees.py
class Node:
    __slots__ = ('left', 'right')
    def __init__(s, l=None, r=None): s.left = l; s.right = r
def make(d):
    if d == 0: return Node()
    return Node(make(d-1), make(d-1))
def check(n):
    if n.left is not None: return 1 + check(n.left) + check(n.right)
    return 1
maxd = 16
long = make(maxd)
for d in range(4, maxd+1, 2):
    it = 1 << (maxd - d + 4)
    total = 0
    for _ in range(it): total += check(make(d))
    print("%d trees of depth %d check %d" % (it, d, total))
print("long lived tree check %d" % check(long))

File: bench/trees.lua
local function make(d) if d == 0 then return {} end return {make(d-1), make(d-1)} end
local function check(t) if t[1] then return 1 + check(t[1]) + check(t[2]) end return 1 end
local maxd = 16
local long = make(maxd)
for d = 4, maxd, 2 do
  local it = 1 << (maxd - d + 4)
  local total = 0
  for i = 1, it do total = total + check(make(d)) end
  print(string.format("%d trees of depth %d check %d", it, d, total))
end
print(string.format("long lived tree check %d", check(long)))
